### DISCUSSION TITLE
why the board NUCLEO_WBA52CG is blocked in modules\hal\stm32\lib\CMakeLists.txt?

### DISCUSSION THREAD

**Discussion by aofrioWeifengLi** [zephyrproject-rtos/zephyr#76860] [Q&A]
there's the following code in modules\hal\stm32\lib\CMakeLists.txt.
 
    if(DEFINED CONFIG_BOARD_NUCLEO_WBA52CG)
      message(FATAL_ERROR "BLE library is not compatible with nucleo_wba52cg")
    endif()

However if using the config STM32WBA52(without NUCLEO) can pass the build, but when using NUCLEO_WBA52CG is blocked. it's not fair.
I removed the code, and build can pass, but BLE doesn't work(I tested the iBeacon sample).  So STM32WBA52 can't be supported on Zephyr?

**Comment by FRASTM** [comment]
Hi @aofrioWeifengLi 

We have already seen such an error (https://github.com/zephyrproject-rtos/hal_stm32/pull/223)
can you try to   'west blobs fetch stm32'

**Reply by aofrioWeifengLi** [reply]
Thanks @FRASTM,
it seems 'west blobs fetch stm32' only download the ble_stack.
And I checked the main branch, the compile error is still in line 23 in
https://github.com/zephyrproject-rtos/hal_stm32/blob/main/lib/CMakeLists.txt

And also, I removed the compiling error, and the compile can pass. (I tested the ibeacon sample), but it's not Advertising the ibeacon package, although nothing wrong is reported...
